core/src/main/java/io/bitsquare/btc/SeedPeersSocks5Dns.java

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright 2011 Micheal Swiggs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.bitsquare.btc;
+
+import com.runjva.sourceforge.jsocks.protocol.Socks5Proxy;
+import com.runjva.sourceforge.jsocks.protocol.SocksSocket;
+
+import org.bitcoinj.net.*;
+import org.bitcoinj.net.discovery.PeerDiscovery;
+import org.bitcoinj.net.discovery.PeerDiscoveryException;
+
+import org.bitcoinj.core.NetworkParameters;
+import javax.annotation.Nullable;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * SeedPeersSocks5Dns resolves peers via Proxy (Socks5) remote DNS.
+ */
+public class SeedPeersSocks5Dns implements PeerDiscovery {
+    private Socks5Proxy proxy;
+    private NetworkParameters params;
+    private InetSocketAddress[] seedAddrs;
+    private InetSocketAddress[] seedAddrsIP;
+    private int pnseedIndex;
+    
+    private InetSocketAddress[] seedAddrsResolved;
+    
+    private static final Logger log = LoggerFactory.getLogger(SeedPeersSocks5Dns.class);
+
+    /**
+     * Supports finding peers by hostname over a socks5 proxy.
+     * 
+     * @param Socks5Proxy proxy the socks5 proxy to connect over.
+     * @param NetworkParameters param to be used for seed and port information.
+     */
+    public SeedPeersSocks5Dns(Socks5Proxy proxy, NetworkParameters params) {
+
+        this.proxy = proxy;
+        this.params = params;
+        this.seedAddrs = convertAddrsString( params.getDnsSeeds(), params.getPort() );
+
+        if( false ) {
+            // This is an example of how .onion servers could be used.  Unfortunately there is presently no way
+            // to hand the onion address (or a connected socket) back to bitcoinj without it crashing in PeerAddress.
+            // note:  the onion addresses should be added into bitcoinj NetworkParameters classes, eg for mainnet, testnet
+            //        not here!
+            this.seedAddrs = new InetSocketAddress[] { InetSocketAddress.createUnresolved( "cajrifqkvalh2ooa.onion", 8333 ),
+                                                       InetSocketAddress.createUnresolved( "bk7yp6epnmcllq72.onion", 8333 )
+            };
```

Comment:
Two sources for .onion addresses:

1. bitcoin-core source code.  See previous comment in #488 where I list them and give a pointer to source.
2. A running bitcoin-core node with onlynet=tor.  I used this to obtain "known good right now" peers for testing a few times.